packages/alea-frontend/pages/api/acl-utils/flattening.ts

```diff
@@ -0,0 +1,52 @@
+import { NextApiResponse } from "next";
+import { executeAndEndSet500OnError } from "../comment-utils";
+import { addSet, getSet } from "./redis-connection-utils";
+
+export class Flattening {
+    private _aclsWantToSearch: Set<string> = new Set<string>();
+    private _aclslisted: string[] = [];
+    private readonly _membersSavePostfix = 'members';
+    private readonly _aclsSavePostfix = 'acls';
+    constructor (private res: NextApiResponse, private readonly _aclMembership: ACLMembership[]) {
+    }
+    public async findMembers(AclId: string): Promise<string[]> {
```

Comment:
I think your code works. But, the code below is shorter and uses just one recursive function. Please consider simplifying if you also think that this will be easier to understand for future readers.

```
_alreadyComputed: string[] = [];
getFlattenedMembers(AclId: string, ancestorChain: string[] = []) {
  if (this._alreadyComputed.includes(AclId)) return await (await getSet(redisName)) as string[];
  const members = this._aclMembership.filter(c=>allAcls.includes(c.parentACLId));
  const output: Set<string> = new Set<string>(); 
  for(const member of members) {
    if(member.memberUserId) {
      output.add(member.memberUserId);
    }else {
       const childACLId = member.memberACLId 
       if (childACLId && !ancestorChain.includes(childACLId) {
         const childMembers = getFlattenedMembers(childACLId, [...ancestorChain, AclId]);
         for(const m of childMembers) output.add(m);
       }
    }
  }
  if (output.size > 0) await addSet(redisName, Array.from(output));
  return Array.from(output);
}
```